src/patches.js

```diff
@@ -1,9 +1,28 @@
-import {each} from "./common"
+import {each, isMap} from "./common"
 
 export function generatePatches(state, basePath, patches, inversePatches) {
-    Array.isArray(state.base)
-        ? generateArrayPatches(state, basePath, patches, inversePatches)
-        : generateObjectPatches(state, basePath, patches, inversePatches)
+    let generatePatchesFn = generatePatchesFromAssigned(
```

Comment:
The flow should be:
```js
generatePatchesFn =
  isArray ? generateArrayPatches
  : isMap ? ...
  // isObject
  : ...
```
And please hoist the functions passed to `generatePatchesFromAssigned` so they can be reused. 👍